File: core/src/main/java/com/ustadmobile/core/contentformats/xapi/Verb.java
package com.ustadmobile.core.contentformats.xapi;

import java.util.Map;

public class Verb {

    private String id;

    private Map<String, String> display;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Map<String, String> getDisplay() {
        return display;
    }

    public void setDisplay(Map<String, String> display) {
        this.display = display;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Verb verb = (Verb) o;

        return id != null ? id.equals(verb.id) : verb.id == null;
    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }
}

File: core/src/commonMain/kotlin/com/ustadmobile/core/contentformats/xapi/Verb.kt
package com.ustadmobile.core.contentformats.xapi

class Verb {

    var id: String? = null

    var display: Map<String, String>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this::class != other::class) return false

        other as Verb

        if (id != other.id) return false
        if (display != other.display) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (display?.hashCode() ?: 0)
        return result
    }


}
